src/vec_buffers.h

```diff
@@ -405,6 +406,12 @@ bool operator==(const Buffers<T>& lhs, const Buffers<T>& rhs)
     return true;
 }
 
+template <typename T>
+void Buffers<T>::swap(unsigned i, unsigned j)
+{
+    std::swap(mem[i], mem[j]);
```

Comment:
Unlikely to matter in our specific case, but it's considered a good practice to write it as:

```C++
void Buffers<T>::swap(unsigned i, unsigned j)
{
    using std::swap;
    swap(mem[i], mem[j]);
}
```

That way, [ADL](https://en.wikipedia.org/wiki/Argument-dependent_name_lookup) can kick in and if a specialized version of `swap` exists it will be used.